### PR TITLE
Add Vision to the Robot

### PR DESCRIPTION
**Pull request by a2aaron**

### CODE REVIEW

File: src/org/usfirst/frc/team1076/robot/Robot.java [hunk 1]
```diff
@@ -43,6 +45,13 @@ public void robotInit() {
 		chooser.addDefault("Default Auto", new ExampleCommand());
 		// chooser.addObject("My Auto", new MyAutoCommand());
 		SmartDashboard.putData("Auto mode", chooser);
+        SmartDashboard.putNumber("Show Vision", 1);
```
Comment (porglezomp) [inline src/org/usfirst/frc/team1076/robot/Robot.java]: We should use `SmarterDashboard` for this!

File: src/org/usfirst/frc/team1076/robot/Robot.java [hunk 2]
```diff
@@ -53,11 +62,18 @@ public void robotInit() {
...
-
-	@Override
+    }
+	
+    int debugCount = 0;
 	public void disabledPeriodic() {
 		Scheduler.getInstance().run();
+		if (debugCount++ % 100 == 0 && SmartDashboard.getNumber("Show Vision", 0) == 1) {
+		    receiver.receive();
+		    if (receiver == null) {
```
Comment (porglezomp) [inline src/org/usfirst/frc/team1076/robot/Robot.java:72]: If receiver was `null`, we would get a NPE on the previous line.

File: src/org/usfirst/frc/team1076/robot/vision/VisionData.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                break;
+            case "error":
+            default:
+                status = VisionStatus.ERROR;
+                errorCount += 1;
+            }
+            heading = json.getInt("heading");
+            range = json.getDouble("range");
+        } catch (JSONException e) { 
+            // Malformed JSON packet, assume no useful data
```
Comment (porglezomp) [inline src/org/usfirst/frc/team1076/robot/vision/VisionData.java]: When we get to this point, we might have already set some data and we should probably clear it.

File: src/org/usfirst/frc/team1076/robot/vision/VisionData.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        }
+    }
+    
+    public void update(String json) {
+    	try {
+    		update(new JSONObject(new JSONTokener(json)));
+    	} catch (JSONException e) {
```
Comment (porglezomp) [inline src/org/usfirst/frc/team1076/robot/vision/VisionData.java:79]: We should perhaps set the status to error?

**Review by porglezomp — CHANGES_REQUESTED** [review]
Definitely fix the NPE bug, the others are probably helpful for robustness too.

File: src/org/usfirst/frc/team1076/robot/Robot.java (hunk 1 above)
Comment (a2aaron) [reply src/org/usfirst/frc/team1076/robot/Robot.java]: We can do this when we merge the SmarterDashboard commits from 2017-Pickup.

File: src/org/usfirst/frc/team1076/robot/Robot.java [hunk 5]
```diff
@@ -46,14 +47,13 @@ public void robotInit() {
 		gamepad = new Gamepad(0);
 		chooser.addDefault("Default Auto", new ExampleCommand());
 		// chooser.addObject("My Auto", new MyAutoCommand());
-		SmartDashboard.putData("Auto mode", chooser);
-        SmartDashboard.putNumber("Show Vision", 1);
-
+    SmarterDashboard.putNumber("Show Vision", 1);
```
Comment (porglezomp) [inline src/org/usfirst/frc/team1076/robot/Robot.java]: This needs to be `putDefaultNumber`

**Review by porglezomp — CHANGES_REQUESTED** [review]
Missed the `putDefaultNumber`

**Review by porglezomp — APPROVED** [review]

**Pull request merged**